src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/CodeFoldingTests.cs

```diff
@@ -84,67 +84,23 @@ static ImmutableArray<Span> GetMissingExpectedSpans(ICollapsible[] outlines, Imm
             }
         }
 
-        private async Task AssertFoldableBlocksAsync(Span[] foldableSpans)
-        {
-            // TODO: REMOVE THIS METHOD AND UPDATE
-            // ALL TESTS TO USE THE STRING OVERLOAD
-
-            var textView = await TestServices.Editor.GetActiveTextViewAsync(HangMitigatingCancellationToken);
-            var outlines = await GetOutlineRegionsAsync(textView);
-
-            //
-            // Built in retry logic because getting spans can take time.
-            //
-            var tries = 0;
-            const int MaxTries = 10;
-            ImmutableArray<Span> missingSpans;
-            while (tries++ < MaxTries)
-            {
-                missingSpans = GetMissingExpectedSpans(outlines, foldableSpans, textView);
-                if (missingSpans.Length == 0)
-                {
-                    return;
-                }
-
-                Thread.Sleep(100);
-            }
-
-            Assert.Empty(missingSpans);
-
-            static ImmutableArray<Span> GetMissingExpectedSpans(ICollapsible[] outlines, Span[] foldableSpans, ITextView textView)
-            {
-                if (outlines.Length < foldableSpans.Length)
-                {
-                    return foldableSpans.ToImmutableArray();
-                }
-
-                var builder = new List<Span>();
-
-                var spans = outlines.Select(o => o.Extent.GetSpan(textView.TextSnapshot).Span).ToImmutableArray();
-                foreach (var foldableSpan in foldableSpans)
-                {
-                    if (!spans.Contains(foldableSpan))
-                    {
-                        builder.Add(foldableSpan);
-                    }
-                }
-
-                return builder.ToImmutableArray();
-            }
-        }
-
-        [IdeFact(Skip = "Coming in a separate checkin")]
+        [IdeFact]
         public async Task CodeFolding_CodeBlock_Default()
         {
             await TestServices.SolutionExplorer.OpenFileAsync(BlazorProjectName, CounterRazorFile, HangMitigatingCancellationToken);
```

Comment:
This is debatable, but it took me a while to work out how the test worked, because it looks like the code in the assert below, is the _input_ not the expected. I wonder if its worth just using `SetTextAsync` to set the file contents, so at least the input and expected are in the same file.

Though I'm sure other tests, probably written by me, commit the same crime.

Reply:
I think having future tests set the stage with their own "starting code" is good practice. The project here is created from the regular Razor project templates, and so if they change their templates it could break these tests which is less than ideal.